test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
-//                 const { execute, read, get } = deployments;
-//                 const { stranger, stranger2, deployer } =
-//                     await getNamedAccounts();
-//                 const vault = randomAddress();
-//                 const vaultGovernance = await get("ERC20VaultGovernance");
-//                 await withSigner(vaultGovernance.address, async (s) => {
-//                     const vaultRegistry = await ethers.getContract(
-//                         "VaultRegistry"
-//                     );
-//                     await vaultRegistry
-//                         .connect(s)
-//                         .registerVault(vault, stranger);
-//                 });
-//                 const nft = await read("VaultRegistry", "vaultsCount");
-//                 await execute(
-//                     "VaultRegistry",
-//                     { from: stranger, autoMine: true },
-//                     "approve",
-//                     stranger2,
-//                     nft
-//                 );
-
-//                 for (const actor of [stranger2, deployer]) {
-//                     await expect(
-//                         execute(
-//                             "VaultRegistry",
-//                             { from: actor, autoMine: true },
-//                             "lockNft",
-//                             nft
-//                         )
-//                     ).to.be.revertedWith(Exceptions.FORBIDDEN);
-//                 }
-//             });
-//         });
-//     });
-// });
+import { expect } from "chai";
+import { ethers, deployments, getNamedAccounts } from "hardhat";
+import {
+    addSigner,
+    now,
+    randomAddress,
+    sleep,
+    sleepTo,
+    withSigner,
+} from "./library/Helpers";
+import { setupDefaultContext, TestContext } from "./library/setup";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signers";
+import { ProtocolGovernance, VaultRegistry } from "./types";
+import { Contract } from "ethers";
+
+type CustomContext = {
+    strategySigner: SignerWithAddress;
+    ownerSigner: SignerWithAddress;
+    newERC20Vault: Contract;
+    newAaveVault: Contract;
+    anotherERC20Vault: Contract;
+    nftERC20: number;
+    nftAave: number;
+    nftAnotherERC20: number;
+    newProtocolGovernance: Contract;
+};
+type DeployOptions = {
+    name?: string;
+    symbol?: string;
+    protocolGovernance?: ProtocolGovernance;
+};
+
+// @ts-ignore
+describe("VaultRegistry", function (this: TestContext<
+    VaultRegistry,
+    DeployOptions
+> &
+    CustomContext) {
+    before(async () => {
+        // @ts-ignore
+        await setupDefaultContext.call(this);
+        this.deploymentFixture = deployments.createFixture(
+            async (_, options?: DeployOptions) => {
+                await deployments.fixture();
+
+                const { address } = await deployments.deploy("VaultRegistry", {
+                    from: this.deployer.address,
```

Comment:
As I understand, deploying new VaultRegistry instead of using the one that already exists, allows to execute VaultGovernance#createVault without triggering VaultRegistry#registerVault
However, I think this behaviour is not quite obvious and deploying a new Vault contract might be a better approach
```solidity
  const { address } = await deploy("AaveVault", {
      from: deployer,
      args: [],
      log: true,
      autoMine: true,
  });
```